### PR TITLE
HOTT-1341: Avoid collisions in meursing additional code values

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

https://transformuk.atlassian.net/browse/HOTT-1341

### What?

I have added/removed/altered:

- [x] Added around_action hook to safely manage meursing additional codes

### Why?

I am doing this because:

- This is required to avoid colliding meursing additional code behaviour

### CODE REVIEW

File: app/controllers/api/v2/commodities_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ module Api
   module V2
     class CommoditiesController < ApiController
       before_action :find_commodity, only: %i[show changes]
-      before_action :set_meursing_additional_code, only: :show
+      around_action :configure_meursing_additional_code, only: :show
```
Comment (alexdesi) [inline app/controllers/api/v2/commodities_controller.rb:5]: I see that this is not just a simple set, but I think it's fine and clearer to keep the name as before: `set_meursing_additional_code`

File: spec/controllers/api/v2/commodities_controller_spec.rb [hunk 2]
```diff
@@ -93,10 +93,32 @@
...
+          :with_meursing_measures,
+          :declarable,
+        )
+      end
+
+      before do
+        Thread.current[:meursing_additional_code_id] = 'foo'
+      end
+
+      it 'sets the value to nil when we are done making the request' do
```
Comment (alexdesi) [inline spec/controllers/api/v2/commodities_controller_spec.rb:112]: small wording improvement:

sets the value to nil when the request completes
or
sets the value to nil when the request  is completed

File: spec/factories/goods_nomenclature_factory.rb [hunk 3]
```diff
@@ -50,6 +50,46 @@
...
+        )
+        # Placeholder meursing measure component - agricultural component
+        create(
+          :measure_component,
+          :with_duty_expression,
+          measure_sid: root_measure.measure_sid,
+          duty_expression_id: '12',
+        )
+
+        root_measure.reload
```
Comment (alexdesi) [inline spec/factories/goods_nomenclature_factory.rb:79]: it seems we are not using `root_measure` this later.

**Review by alexdesi — APPROVED** [review]
It looks good, 
I added minor comments.

File: app/controllers/api/v2/commodities_controller.rb (hunk 1 above)
Comment (willfish) [reply app/controllers/api/v2/commodities_controller.rb:5]: I'm actually following the same approach as the time machine, basically. I could extract this into a MuersingConfigurator or equivalent at some point

**Pull request merged**